pkg/autoscaler/multiscaler_test.go

```diff
@@ -14,150 +14,153 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package autoscaler_test
+package autoscaler
 
 import (
 	"context"
+	"errors"
 	"fmt"
 	"sync"
 	"testing"
 	"time"
 
 	. "github.com/knative/pkg/logging/testing"
 	kpa "github.com/knative/serving/pkg/apis/autoscaling/v1alpha1"
-	"github.com/knative/serving/pkg/autoscaler"
-	"k8s.io/apimachinery/pkg/api/errors"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 const (
-	testRevision  = "test-revision"
-	testNamespace = "test-namespace"
-	testKPAKey    = "test-namespace/test-revision"
+	tickInterval = 5 * time.Millisecond
+	tickTimeout  = 50 * time.Millisecond
 )
 
+// watchFunc generates a function to assert the changes happening in the multiscaler.
+func watchFunc(ctx context.Context, ms *MultiScaler, metric *Metric, desiredScale int, doneCh chan struct{}, errCh chan error) func(key string) {
```

Comment:
nit: we could probably have just the errCh (maybe call it watchCh or similar) and use nil vs non-nil to distinguish between done and error which would remove many LOC